src/Integration.UnitTests/Suppression/SuppressedIssuesFileSynchronizerTests.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            var suppressedIssuesMonitor = new Mock<ISuppressedIssuesMonitor>();
+            var activeSolutionBoundTracker = new Mock<IActiveSolutionBoundTracker>();
+            var suppressedIssuesFileStorage = new Mock<ISuppressedIssuesFileStorage>();
+
+            activeSolutionBoundTracker.Setup(x => x.CurrentConfiguration).Returns(BindingConfiguration.Standalone);
+
+            CreateTestSubject(suppressedIssuesMonitor: suppressedIssuesMonitor.Object,
+                activeSolutionBoundTracker: activeSolutionBoundTracker.Object,
+                suppressedIssuesFileStorage: suppressedIssuesFileStorage.Object);
+
+            suppressedIssuesMonitor.Raise(x=> x.SuppressionsUpdateRequested += null, EventArgs.Empty);
+
+            activeSolutionBoundTracker.Verify(x=> x.CurrentConfiguration, Times.Once);
+            suppressedIssuesFileStorage.Invocations.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)] // should update storage even when there are no issues
+        public void OnSuppressionsUpdateRequested_ConnectedMode_StorageUpdated(bool hasIssues)
+        {
+            var suppressedIssuesMonitor = new Mock<ISuppressedIssuesMonitor>();
+            var activeSolutionBoundTracker = new Mock<IActiveSolutionBoundTracker>();
+            var suppressedIssuesFileStorage = new Mock<ISuppressedIssuesFileStorage>();
+            var suppressedIssuesProvider = new Mock<ISonarQubeIssuesProvider>();
+
+            var configuration = CreateBindingConfiguration("some project key");
+            activeSolutionBoundTracker.Setup(x => x.CurrentConfiguration).Returns(configuration);
+
+            var issues = hasIssues ? new[] { CreateSonarQubeIssue(), CreateSonarQubeIssue() } : Array.Empty<SonarQubeIssue>();
+            suppressedIssuesProvider.Setup(x => x.GetAllSuppressedIssues()).Returns(issues);
+
+            CreateTestSubject(suppressedIssuesMonitor: suppressedIssuesMonitor.Object,
+                activeSolutionBoundTracker: activeSolutionBoundTracker.Object,
+                suppressedIssuesFileStorage: suppressedIssuesFileStorage.Object,
+                suppressedIssuesProvider: suppressedIssuesProvider.Object);
+
+            suppressedIssuesMonitor.Raise(x => x.SuppressionsUpdateRequested += null, EventArgs.Empty);
+
+            suppressedIssuesFileStorage.Verify(x=> x.Update("some project key", issues), Times.Once);
+        }
+
+        private SonarQubeIssue CreateSonarQubeIssue()
+        {
+            return new SonarQubeIssue(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(),
+                Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), false,
+                SonarQubeIssueSeverity.Blocker, new DateTimeOffset(), new DateTimeOffset(), null, null);
+        }
+
+        private static BindingConfiguration CreateBindingConfiguration(string projectKey)
```

Comment:
Might be better to call this `CreateBoundConfiguration`/`CreateConnectedConfiguration` since it's hard-coded to return use connected mode.